bot/cogs/cogs.py

```diff
@@ -0,0 +1,194 @@
+
+        if full_cog == "bot.cogs.cogs":
+            embed.description = "You may not unload the cog management cog!"
+        elif full_cog in self.bot.extensions:
+            try:
+                self.bot.unload_extension(full_cog)
+            except Exception as e:
+                embed.description = f"Failed to unload cog: {cog}\n\n```{e}```"
+            else:
+                embed.description = f"Cog unloaded: {cog}"
+                embed.colour = Colour.green()
+        else:
+            embed.description = f"Cog {cog} is not loaded"
+
+        await ctx.send(embed=embed)
+
+    @command(name="cogs.reload()", aliases=["cogs.reload", "reload_cog"])
+    @with_role(MODERATOR_ROLE, ADMIN_ROLE, OWNER_ROLE, DEVOPS_ROLE)
+    async def reload_command(self, ctx: Context, cog: str):
+        cog = cog.lower()
+
+        embed = Embed()
+        embed.colour = Colour.red()
+
+        embed.set_author(
+            name="Python Bot (Cogs)",
+            url="https://github.com/discord-python/bot",
+            icon_url="https://raw.githubusercontent.com/discord-python/branding/master/logos/logo_circle.png"
+        )
+
+        if cog in self.cogs:
+            full_cog = self.cogs[cog]
+        elif "." in cog:
+            full_cog = cog
+        else:
+            full_cog = None
+            embed.description = f"Unknown cog: {cog}"
+
+        if full_cog == "bot.cogs.cogs":
+            embed.description = "You may not reload the cog management cog!"
+        elif full_cog in self.bot.extensions:
+            try:
+                self.bot.unload_extension(full_cog)
+                self.bot.load_extension(full_cog)
+            except Exception as e:
+                embed.description = f"Failed to reload cog: {cog}\n\n```{e}```"
+            else:
+                embed.description = f"Cog reload: {cog}"
+                embed.colour = Colour.green()
+        else:
+            embed.description = f"Cog {cog} is not loaded"
+
+        await ctx.send(embed=embed)
+
+    @command(name="cogs.get_all()", aliases=["cogs.get_all", "get_cogs", "get_all_cogs", "cogs"])
+    @with_role(MODERATOR_ROLE, ADMIN_ROLE, OWNER_ROLE, DEVOPS_ROLE)
+    async def list_command(self, ctx: Context):
+        embed = Embed()
+        lines = []
+        cogs = {}
+
+        embed.colour = Colour.blurple()
+        embed.set_author(
+            name="Python Bot (Cogs)",
+            url="https://github.com/discord-python/bot",
+            icon_url="https://raw.githubusercontent.com/discord-python/branding/master/logos/logo_circle.png"
+        )
+
+        for key, value in self.cogs.items():
+            if "." not in key:
+                continue
+
+            if key in self.bot.extensions:
+                cogs[key] = True
+            else:
+                cogs[key] = False
+
+        for key in self.bot.extensions.keys():
+            if key in self.cogs:
+                continue
+
+            cogs[key] = True
```

Comment:
Could we have this line in the `if` clause, and change it to a `if key not in self.cogs`?